daemon/container_linux.go

```diff
@@ -306,6 +307,24 @@ func populateCommand(c *Container, env []string) error {
 	return nil
 }
 
+func mergeDevices(defaultDevices, userSpecifiedDevices []*configs.Device) []*configs.Device {
+	if len(userSpecifiedDevices) == 0 {
+		return defaultDevices
+	}
+
+	// Only add those default devices, which are not specified by the user.
+	// Otherwise, the user specified devices end up without permissions.
+	var devs []*configs.Device
+	for _, defaultDevice := range defaultDevices {
+		for _, userSpecified := range userSpecifiedDevices {
+			if defaultDevice.Path != userSpecified.Path {
```

Comment:
I'm confused, but maybe it's because I don't know go. Does this add a copy of each default device for every user specified device which doesn't have the same path as the default device? If so, that would be wrong. But then again, I really don't know golang so...